symbolic-unwind/src/evaluator/mod.rs

```diff
@@ -61,49 +62,40 @@ pub struct Evaluator<'memory, A, E> {
     /// operations will fail.
     memory: Option<MemoryRegion<'memory>>,
 
-    /// A map containing the values of constants.
-    ///
-    /// Trying to use a constant that is not in this map will cause evaluation to fail.
-    constants: BTreeMap<Constant, A>,
-
-    /// A map containing the values of variables.
+    /// A map containing the values of registers.
     ///
-    /// Trying to use a variable that is not in this map will cause evaluation to fail.
+    /// Trying to use a register that is not in this map will cause evaluation to fail.
     /// This map can be modified by the [`assign`](Self::assign) and
     ///  [`process`](Self::process) methods.
-    variables: BTreeMap<Variable, A>,
+    registers: BTreeMap<Register, A>,
 
     /// The endianness the evaluator uses to read data from memory.
     endian: E,
+
+    cache: BTreeMap<Expr<A>, A>,
 }
 
-impl<'memory, A, E> Evaluator<'memory, A, E> {
+impl<'memory, A: Ord, E> Evaluator<'memory, A, E> {
     /// Creates an Evaluator with the given endianness, no memory, and empty
     /// constant and variable maps.
     pub fn new(endian: E) -> Self {
         Self {
             memory: None,
-            constants: BTreeMap::new(),
-            variables: BTreeMap::new(),
+            registers: BTreeMap::new(),
             endian,
+            cache: BTreeMap::new(),
         }
     }
 
-    /// Sets the evaluator's memory to the given `MemoryRegion`.
+    /// Sets the evaluator's [memory](Self::memory) to the given `MemoryRegion`.
```

Comment:
This is self-referential, since it just links to the method you're documenting. The field is private.
Same below for `registers`.

Reply:
D'oh.